2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,994 @@
+  Execution Environment (TEE) technologies and standards
+  - With Confidential Computing being the protection of data in use by
+  performing computation in a hardware-based, attested Trusted Execution
+  Environment.
+
+---
+
+# TEE locations
+
+.right-column45[
+.center[
+
+**CPU based TEE**
+
+<img src="/img/TEE_CPU_based.svg" width="250px">
+
+]
+]
+
+.left-column70[
+### Examples
+
+- Arm Trustzone
+- Intel SGX (Software Guard Extensions)
+- AMD SEV (Secure Encrypted Virtualization)
+
+]
+
+---
+
+# TEE locations
+
+.right-column45[
+.center[
+
+**Coprocessor Based TEE (One SoC)**
+
+<img src="/img/TEE_CoCPU_based.svg" width="250px">
+
+]
+]
+
+.left-column70[
+### Examples
+
+- Integrated TPM (Trusted Platform Module)
+- Apple Secure Enclave
+
+]
+
+???
+
+- ARM Cortex-M based Secure Subsystem
+- Shared resources
+  - Power, memory, caches, buses, clock signals
+  - Larger vector of attack
+- Hardware/firmware vulnerabilities could be exploited
+
+---
+
+# TEE locations
+
+.right-column45[
+.center[
+
+**Coprocessor Based TEE (External)**
+
+<img src="/img/TEE_SepCPU_based.svg" width="240px">
+
+]
+]
+
+.left-column70[
+### Examples
+
+- Discrete TPM (Trusted Platform Module)
+- Apple T2 Chip
+- Google Titan M
+
+]
+
+---
+
+# Normal vs Secure World - Arm Cortex-A
+
+.left-column50[
+<br>
+
+### Overview
+
+<br>
+<br>
+<br>
+<br>
+<br>
+<br>
+<br>
+
+### Exception Levels
+
+
+]
+
+.right-column50[
+<img src="/img/TEE_ARM_Cortex-a.svg" height="220px" style="margin-left:-120px; margin-top:-10px">
+`
+<img src="/img/TEE_ARM_Cortex-a_exception_levels.svg" height="190px" style="margin-left:-120px; margin-top:40px">
+]
+
+???
+
+- Arm TrustZone
+- Arm operates on Exception Levels which roughly correlate with x86 protection
+rings
+- High-level overview
+  + Trusted Applications are also protected from each other
+- The Secure Partition Manager is optional, just as you can run a Rich OS without a
+  hypervisor you can run a Trusted OS without the Secure Partition Manager
+- Worth noting that the Secure Partition Manager can be also implemented in EL3
+- You can see how the data goes through Secure Monitor which with conjuction
+with Memory Managment Unit
+  - Data goes from Rich OS via "Secure Monitor Call (SMC)"
+- Secure montior also saves the states of Normal and secure world when switching
+contexts to be able to restore them when switching back
+
+---
+
+# Normal vs Secure Worlds - Arm Cortex-M
+
+.center[ <img src="/img/TEE_ARM_Cortex-m.svg" height="250px"> ]
+
+???
+
+- The division between worlds in Armv8-M is memory map-based
+  + The transitions take place automatically via exception handling
+  + This means that, when running code from the secure memory, the processor
+    state is secure, and, when running code from non-secure memory, the processor
+    state is non-secure 
+  + This also excludes the monitor mode and the need for any secure monitor
+    software
```

Comment:
```suggestion
    firmware
```

Reply:
You're right. [addressed](https://github.com/3mdeb/conferences/pull/36/commits/2b80f55523912f39e906cf451d26952f4b0b6dda)